src/beanmachine/ppl/diagnostics/tools/utils/base.py

```diff
@@ -0,0 +1,85 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+"""Base class for diagnostic tools of a Bean Machine model."""
+
+import re
+from abc import ABC, abstractmethod
+from typing import TypeVar
+
+from beanmachine.ppl.diagnostics.tools import JS_DIST_DIR
+from beanmachine.ppl.diagnostics.tools.utils import plotting_utils
+from beanmachine.ppl.diagnostics.tools.utils.model_serializers import serialize_bm
+from beanmachine.ppl.inference.monte_carlo_samples import MonteCarloSamples
+from bokeh.embed import file_html
+from bokeh.models import Model
+from bokeh.resources import INLINE
+from IPython.display import display, HTML
+
+
+T = TypeVar("T", bound="Base")
+
+
+class Base(ABC):
```

Comment:
Can we have a more descriptive class name or a more descriptive package name? `utils.base` is a bit hard to understand without reading through example usage

Reply:
Module renamed to `diagnostic_tool_base.py` and the class in it is renamed to `DiagnosticToolBaseClass`.